File: Java/RemoveDuplicatesfromSortedArray.java
//
//  LeetCode
//  Created by Abdelrahman Nasser
//

class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums.length == 0)
            return 0;
        
        int lastSortedIndex = 1; 

        for(int i = 0; i < nums.length - 1; i++) {
            
            if(nums[i] < nums[i + 1]) {
              nums[lastSortedIndex] = nums[i + 1];
              lastSortedIndex++;
            }
        }
        return lastSortedIndex;
    }
}

File: Swift/RemoveDuplicatesfromSortedArray.swift
//
//  LeetCode
//  Created by Abdelrahman Nasser
//

class Solution {
    func removeDuplicates(_ nums: inout [Int]) -> Int {
        var count = 1
        var currentValue = nums.first ?? 0
        var lastSortedIndex = 1
        for (index, value) in nums.enumerated() {
            if currentValue < value {
                currentValue = value
                nums[lastSortedIndex] = value
                lastSortedIndex+=1
                count += 1
            }
        }
        return count
    }
}
